controllers/networking-calico/pkg/controller/status.go

```diff
@@ -0,0 +1,59 @@
+package controller
+
+import (
+	"context"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+
+	calicov1alpha1 "github.com/gardener/gardener-extensions/controllers/networking-calico/pkg/apis/calico/v1alpha1"
+	calicov1aplha1 "github.com/gardener/gardener-extensions/controllers/networking-calico/pkg/apis/calico/v1alpha1"
+	extensionscontroller "github.com/gardener/gardener-extensions/pkg/controller"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/retry"
+)
+
+func (a *actuator) updateProviderStatus(
+	ctx context.Context,
+	network *extensionsv1alpha1.Network,
+	config *calicov1aplha1.NetworkConfig,
+) error {
+	status, err := a.ComputeNetworkStatus(config)
+	if err != nil {
+		return err
+	}
+
+	return extensionscontroller.TryUpdateStatus(ctx, retry.DefaultBackoff, a.client, network, func() error {
+		network.Status.ProviderStatus = &runtime.RawExtension{Object: status}
+		network.Status.LastOperation = extensionscontroller.LastOperation(gardencorev1alpha1.LastOperationTypeReconcile,
+			gardencorev1alpha1.LastOperationStateSucceeded,
+			100,
+			"Calico was configured successfully")
+		return nil
+	})
+}
+
+func (a *actuator) ComputeNetworkStatus(networkConfig *calicov1alpha1.NetworkConfig) (*calicov1alpha1.NetworkConfigStatus, error) {
+	var (
+		status = &calicov1alpha1.NetworkConfigStatus{
+			TypeMeta: StatusTypeMeta,
+		}
+	)
+	// TODO: return health-check results for calico components as well network connectivity tests pass result
```

Comment:
What about this TODO? Why should we return the 'health status" in the calico provider status? I don't think this makes sense, we should rather have `.status.conditions[]` on the `Network` resource itself.

Reply:
because it can be specific to the plugin, component names for each plugin change.

Reply:
That's clear, but I don't get the motivation to expose this kind of information in the provider status at all. As said, we will have `.status.conditions[]` which will indicate the health of the deployed resources, so why duplicating it to the provider status?